news/12696-do-not-set-context-CONDA_PREFIX

```diff
@@ -0,0 +1,19 @@
+### Enhancements
+
+* Do not mock `$CONDA_PREFIX` when `--name` or `--prefix` is provided. (#12696)
```

Comment:
I would probably mention something about the introduction of `Template` to parse environment variables. Could be helpful with tracking down bugs with the release notes in the future.